src/configuration.ts

```diff
@@ -25,7 +25,8 @@ import { arrayify, hasOwnProperty, stripComments } from "./utils";
 
 export interface IConfigurationFile {
     /**
-     * The severity that is applied to rules in _this_ config with `severity === "default"`.
+     * The severity that is applied to rules in this config file as well as rules
+     * in any inherited config files which have their severity set to "default".
      * Not inherited.
      */
     defaultSeverity?: RuleSeverity;
```

Comment:
It is used on lines 70 and 78:

    export const DEFAULT_CONFIG: IConfigurationFile = {
        defaultSeverity: "error",

    export const EMPTY_CONFIG: IConfigurationFile = {
        defaultSeverity: "error",

Reply:
Yeah, I know. But the value is never read. In addition `parseConfigFile` and `extendConfigurationFile` simply don't set this property. That means you can basically never use it.
Since this could (very unlikely) break some users of the API, we should keep it around until the next major version. So this is not immediately actionable. I'll open a separate PR for that.

While you are already touching this code, can you please add `@deprecated property is never set` in the JSDoc comment? This could lead to lint errors on the two lines you mentioned above. You can just disable the error with `// tslint:disable-line:deprecation`.